packages/app/src/cli/services/function/replay.ts

```diff
@@ -92,21 +92,36 @@ async function getFunctionRunData(functionRunsDir: string, functionHandle: strin
       return splitFilename[3] === 'extensions' && splitFilename[4] === functionHandle
     })
     .reverse()
-  const latestFunctionRunFileNames = allFunctionRunFileNames.slice(0, LOG_SELECTOR_LIMIT)
-  const functionRunFilePaths = latestFunctionRunFileNames.map((functionRunFile) =>
-    joinPath(functionRunsDir, functionRunFile),
-  )
 
-  const functionRunData = await Promise.all(
-    functionRunFilePaths.map(async (functionRunFilePath) => {
-      const fileData = await readFile(functionRunFilePath)
-      const parsedData = JSON.parse(fileData)
-      return {
-        ...parsedData,
-        identifier: getIdentifierFromFilename(functionRunFilePath),
-      }
-    }),
-  )
+  let functionRunData: FunctionRunData[] = []
+  for (let i = 0; i < allFunctionRunFileNames.length; i += LOG_SELECTOR_LIMIT) {
+    const currentFunctionRunFileNameChunk = allFunctionRunFileNames.slice(i, i + LOG_SELECTOR_LIMIT)
+    const functionRunFilePaths = currentFunctionRunFileNameChunk.map((functionRunFile) =>
+      joinPath(functionRunsDir, functionRunFile),
+    )
+
+    // eslint-disable-next-line no-await-in-loop
+    const functionRunDataFromChunk = await Promise.all(
+      functionRunFilePaths.map(async (functionRunFilePath) => {
+        const fileData = await readFile(functionRunFilePath)
+        const parsedData = JSON.parse(fileData)
+        return {
+          ...parsedData,
+          identifier: getIdentifierFromFilename(functionRunFilePath),
+        }
+      }),
+    )
+
+    functionRunData = functionRunData.concat(
+      functionRunDataFromChunk.filter((run) => {
+        return run.payload.input !== null
+      }),
+    )
+
+    if (functionRunData.length >= LOG_SELECTOR_LIMIT) {
+      break
+    }
+  }
```

Comment:
Alternatives considered:

Read the entire directory, filter out empty inputs, take the first 100
I've avoided this approach since reading the entire directory could be a slow operation depending on the directories size
Ignore the issue and accept that replay lists could be small or empty in the event that a significant portion of the last 100 events have no input
I think this would be a bad look for the product, even if it was a rare occurence.
Indicate if a log is 'replay'able in the filename, so that we can filter them out at the same time as we filter out extension handles
I've avoided this because the filename is already fairly large

Reply:
>If the function has no input, and thus isn't replayable, is there any other data that makes it worth storing in the temp directory?

There would be an associated error message explaining what went wrong when we tried to produce the input which could be valuable

Reply:
Though if the only purpose of saving the log to a file is to enable replays, and not for future reference, then I would agree that is the preferred approach.

Reply:
Thoughts @nickwesselman?

Reply:
Saving them is also what enables you to `Open log file` during `shopify app dev`... I think we should keep these.

Is the better overall solution here to provide some ability to preview the execution before running it? Do we have something captured around the `Press 'd' for details` concept from the DX spec?

Reply:
>Do we have something captured around the Press 'd' for details concept from the DX spec?

Just added a ticket: https://github.com/Shopify/shopify-functions/issues/254, since I know we'll want this anyway.

>Is the better overall solution here to provide some ability to preview the execution before running it?

Given that we have no defined behaviour for running a function with no input, I don't think it makes much sense to expose the run as an option- even if we did expose a way for the partners to see ahead of time that the input was empty.